### PR TITLE
Skip targets with unsupported platforms instead of throwing an error

### PR DESCRIPTION
**Pull request by fortmarek**
Resolves https://github.com/tuist/tuist/issues/6030

### Short description 📝

We should only skip targets in the `PackageInfoMapper` that have only unsupported platforms as a condition, such as `linux`.

### How to test the changes locally 🧐

- You should be able to generate and build the `app_with_spm_dependencies` fixture. The error was caused by [this](https://github.com/realm/SwiftLint/blob/main/Package.swift#L59) SwiftLint target that has a condition with `linux` only

### Contributor checklist ✅

- [x] The code has been linted using run `mise run lint:fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [x] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase
- [x] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

**Review by danieleformichelli — APPROVED** [review]

File: Sources/TuistLoader/SwiftPackageManager/PackageInfoMapper.swift [hunk 1]
```diff
@@ -486,6 +486,11 @@ public final class PackageInfoMapper: PackageInfoMapping {
                 do {
                     let condition = try ProjectDescription.PlatformCondition.from(setting.condition)
```
Comment (waltflanagan) [inline Sources/TuistLoader/SwiftPackageManager/PackageInfoMapper.swift:487]: This doesnt throw anymore, so we should be able to update this callsite and remove the `do {} catch {}` also

**Comment by waltflanagan** [comment]
I'm not able to find where this ended up being able to throw out in a way that would end up failing things.  Is the code that doesnt handle the `throw`  in tuist cloud?

**Comment by fortmarek** [comment]
> I'm not able to find where this ended up being able to throw out in a way that would end up failing things. Is the code that doesnt handle the throw in tuist cloud?

No, `tuist generate` would throw when mapping the `PackageInfo` from https://github.com/realm/SwiftLint

File: Sources/TuistLoader/SwiftPackageManager/PackageInfoMapper.swift (hunk 1 above)
Comment (fortmarek) [reply Sources/TuistLoader/SwiftPackageManager/PackageInfoMapper.swift:487]: oh yeah, good point

**Pull request merged**